### PR TITLE
Fix Apple Silicon compatibility and general cleanup

### PR DESCRIPTION
**Pull request by ealeksandrov**
## Description

With https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/610 (`GoogleSignIn` update) merged we don't have any stale dependencies holding us back from arm64 simulator compatibility.

This PR:
- removes settings to exclude arm64 in project and Podfile
- removes static framework setting from Podspec
- converts code to Swift 5
- update minimum requirement to iOS 13
- removes unused `OHHTTPStubs` dev dependency
- updates `OCMock` dev dependency
- fixes all new Xcode warnings

Tested on Intel, M1 in Rosetta mode, M1 native.

## Test

1. Clone this repo, install gems and pods (instructions in README), open project, build and run unit tests.
2. Try updated dependency with WC-iOS or WP-iOS app. (Switch to test version by using `try/apple-silicon-updated` branch for `WordPressAuthenticator-iOS` in Podfile.)

Additional note for WC-iOS + M1 users: if you want to test on arm64 simulator - hack `StripeTerminal` (like described in p91TBi-5tl-p2). It's not needed for device build or Rosetta mode.

### CODE REVIEW

**Comment by ealeksandrov** [comment]
There is an issue on `pod lib lint` check in CI (reproduced locally) that I don't know how to fix yet:

```
> WordPressKit/AccountSettingsRemote.swift:
> import WordPressShared
error: compiling for iOS 11.0, but module 'WordPressShared' has a minimum deployment target of iOS 13.0:
```

`WPAuth` has iOS 13 target in this branch. One of dependencies is `WordPressShared` 1.16.1 that targets iOS 13. But `WordPressKit` 4.39.0 targets iOS 11 and CocoaPods builds it without bumping to iOS 13.

Easiest way to reproduce:
1. Pull the branch
2. Remove post_install script in Podfile:
https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/331a631660b00e93b82c5e6ee6c007caa12d20eb/Podfile#L50-L63
3. `bundle exec pod update`
4. Try building the workspace

While we can hack deployment target with `post_install` on local setup - it doesn't apply to `pod lib lint`.

**Comment by mokagio** [comment]
@ealeksandrov thank you for taking this on board 🙌 

WordPress iOS [targets iOS 13](https://github.com/wordpress-mobile/WordPress-iOS/blob/016d6a58f179fc1906b152297248acd2850aff67/Podfile#L14). Maybe we could address this by bumping the deployment target for [WordPressKit](https://github.com/wordpress-mobile/WordPressKit-iOS/) to 13.0, too?

**Review by dvdchr — COMMENTED** [review]
Done with my first pass of review.
- Code changes look good to me.
- Build & test passes on my Intel and M1 machines. 👍 

Next, I'll take a look at integrating this to WP-iOS and running it in both my Intel & M1 machines.

> Maybe we could address this by bumping the deployment target for WordPressKit to 13.0, too?

+1 on this. 🙂

File: Podfile.lock [hunk 1]
```diff
@@ -25,22 +25,7 @@ PODS:
...
-    - OHHTTPStubs/NSURLSession
-    - OHHTTPStubs/OHPathHelpers
-  - OHHTTPStubs/JSON (8.0.0):
-    - OHHTTPStubs/Core
-  - OHHTTPStubs/NSURLSession (8.0.0):
-    - OHHTTPStubs/Core
-  - OHHTTPStubs/OHPathHelpers (8.0.0)
-  - OHHTTPStubs/Swift (8.0.0):
-    - OHHTTPStubs/Default
+  - OCMock (3.8.1)
```
Comment (dvdchr) [inline Podfile.lock:28]: 👍 !! Adding context to this, support for ARM-based Macs is introduced in [OCMock 3.8](https://github.com/erikdoe/ocmock/releases/tag/v3.8).

**Comment by ealeksandrov** [comment]
@mokagio @dvdchr thank you for the suggestion! I agree with alignment of deployment target everywhere.
The PR for WordPressKit is ready: https://github.com/wordpress-mobile/WordPressKit-iOS/pull/430.

**Comment by ealeksandrov** [comment]
CI is green, lint passed 🎉

File: Podfile [hunk 2]
```diff
@@ -43,8 +43,6 @@ target 'WordPressAuthenticatorTests' do
   project 'WordPressAuthenticator.xcodeproj'
   wordpress_authenticator_pods
 
-  pod 'OHHTTPStubs', '8.0.0'
-  pod 'OHHTTPStubs/Swift', '8.0.0'
```
Comment (AliSoftware) [inline Podfile:47]: 😢  goodbye my baby 👋 😄

**Review by jkmassel — APPROVED** [review]

**Review by dvdchr — APPROVED** [review]

**Review by mokagio — APPROVED** [review]
Hey @ealeksandrov! Sorry for not following up on your progress on this yesterday after you merged the WordPress Kit iOS 13 bump.

I wanted to checkout the Pod on my M1 Mac Mini and see if it all built fine, but I run out of time, both yesterday and today.

Other folks have approved this already, and I agree. We'll deal with M1 issues if they arise 👍

**Comment by ealeksandrov** [comment]
@mokagio no worries! + @jkmassel @dvdchr thanks for everyones reviews and suggestions! 🚀

**Pull request merged**